### PR TITLE
Fix #6569: Receiving address missing for Sign and Send SPL Token tx

### LINKED ISSUE #6569: Sign and Send transaction doesn't show the receiving address
### Description: 
Sign and Send transaction doesn't show the receiving address

### Steps to Reproduce 
  1. Connect to Solana Dapp
  2. Click on Sign and Send transaction
  3. Receiving address is not shown

**Actual result:** 
image

**Expected result:**
Should show receiving address

**Reproduces how often:** [Easily reproduced, Intermittent Issue]
Easy

**Brave Version:** 
1.46 (22.12.6.11)

**Device details:** 
All 

**Website problems only:**
- did you check with Brave Shields down? NA
- did you check in Safari/Firefox (WkWebView-based browsers)? NA

### Additional Information
cc: @nuo-xu @StephenHeaps @brave/sec-team

### PR DESCRIPTION
**Pull request by StephenHeaps**
## Summary of Changes
- Resolve missing receiving address for Sign and Send SPL Token transaction

This pull request fixes #6569

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`
- [x] New or updated UI has been tested across:
  - [x] Light & dark mode
  - [x] Different size classes (iPhone, landscape, iPad)
  - [x] Different dynamic type sizes

## Test Plan:
1. Open test dapp site: https://pwgoom.csb.app/
2. Connect and tap 'Sign and Send SPL Token Transaction'
3. Verify to address is displayed correctly
    - Note: May differ to desktop until https://github.com/brave/brave-browser/issues/27187 is resolved.
    - Since this is an SPL Token transaction, we need to get the receiving address from the Solana Instructions. In this case it is a [Token Program - Transfer instruction type](https://docs.rs/spl-token/latest/src/spl_token/instruction.rs.html#97-110) so the `The destination account.` should be the second address displayed in the 'Details' tab of the Transaction Confirmation.

## Screenshots:

https://user-images.githubusercontent.com/5314553/206067552-2aeb7f55-fc40-4b32-84e9-15995e7ce93e.mp4

## Reviewer Checklist:

- [ ] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [ ] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [ ] Adequate unit test coverage exists to prevent regressions.
- [ ] Adequate test plan exists for QA to validate (if applicable).
- [ ] Issue and pull request is assigned to a milestone (should happen at merge time).

### CODE REVIEW

**Review by nuo-xu — APPROVED** [review]

**Comment by kdenhartog** [comment]
The code here looks good, but I want to double check what happens here when the to/from address match. I'm thinking we just show a single alias name for this which would match what we're doing on desktop and should be good enough to address [this concern](https://mirror.xyz/x-explore.eth/cL3d_CyNujXq8XY7ueP4omNXx_IY1EG5Dz0FD0vJ90M)

**Comment by StephenHeaps** [comment]
> The code here looks good, but I want to double check what happens here when the to/from address match. I'm thinking we just show a single alias name for this which would match what we're doing on desktop and should be good enough to address [this concern](https://mirror.xyz/x-explore.eth/cL3d_CyNujXq8XY7ueP4omNXx_IY1EG5Dz0FD0vJ90M)

@kdenhartog When the from and to address match, we only show the single Blockie & account address without the arrow to match desktop.
https://github.com/brave/brave-ios/blob/development/Sources/BraveWallet/Crypto/Transactions/TransactionHeader.swift#L28-L34

This is testable using the `Sign and Send Transaction` on pwgoom.csb.app which is [a transfer transaction from your connected pubkey to your connected pubkey](https://github.com/darkdh/solana-provider-test/blob/main/src/App.tsx#L109-L112).

I had another change to only show the from address when the to address is empty but looks like I missed pushing that change up here. This would occur if all instructions in a transaction were unknown (or do not contain `to_account` key). I'll push that fix up.

File: Sources/BraveWallet/Extensions/SolanaInstruction+Extensions.swift [hunk 1]
```diff
@@ -37,6 +37,18 @@ extension BraveWallet.SolanaInstruction {
     }
     return Strings.Wallet.solanaUnknownInstructionName
   }
+  
+  /// Returns the `to_account` pubkey for the instruction if available
+  var toPubkey: String? {
+    guard let index = decodedData?.accountParams.firstIndex(where: { $0.name == "to_account" }) else { return nil }
+    return accountMetas[safe: index]?.pubkey
```
Comment (stoletheminerals) [inline Sources/BraveWallet/Extensions/SolanaInstruction+Extensions.swift:44]: Could there be an edge case where `to_account` exists but its `pubkey` doesn't? I tried to test it dynamically by deleting `transaction.instructions[0].keys[0].pubkey` from the transaction body, but have a hard time manually crafting Solana transactions. Maybe we should check if `pubkey` exists in `guard` close just in case.

Comment (StephenHeaps) [reply Sources/BraveWallet/Extensions/SolanaInstruction+Extensions.swift:44]: There is a 1:1 relationship between `accountMetas` and `decodedData.accountParams` so this shouldn't occur, but _if_ there was no account available then `accountMetas[safe: index]?` would fail to unwrap and this computed property would return nil so we would only display the from address & blockie in the transaction header.

The `safe:` helper is found here (if the index does not exist, return nil): https://github.com/brave/brave-ios/blob/development/Sources/Shared/Extensions/ArrayExtensions.swift#L57-L62

**Review by kdenhartog — APPROVED** [review]
Looks good to me
